### PR TITLE
edited setup guide :bento:

### PR DESCRIPTION
**Pull request by heyAyushh**
Added more comprehensive guide with build dependencies.

### CODE REVIEW

**Comment by heyAyushh** [comment]
is there a way we can preview the document on the site?
I have added some images, also checked the preview on markdown VSCode extension.

File: content/guides/getstarted/setup-local-development.md [hunk 1]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
 - how to create a Solana wallet for developing
 - how to airdrop SOL tokens for your wallet
 
-## Windows users only
+If you are very new to development in general, or have just tried EVM based blockchains or just have fiddled with Nodejs and reactJS, your machine is not yet ready to help you code on solana.
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:41]: Minor but remember to capitalize React, Node.js, Windows, etc. macOS is just macOS though according to Apple.

File: content/guides/getstarted/setup-local-development.md [hunk 2]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
...
-Once you've installed both Windows Subsystem for Linux and Windows Terminal,
-open Windows Terminal and start a new tab for your Linux distribution (e.g.
-'Ubuntu') using the `wsl` command.
+This guide has seven sections
 
-Then continue with this guide, running each of the terminal commands within this
-Linux terminal.
+1. Install Dependencies
+2. Install rust and cargo toolchain
+3. Install Solana cli
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:52]: CLI

File: content/guides/getstarted/setup-local-development.md [hunk 3]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
...
+## 1. Installing Dependencies
 
-To interact with the Solana clusters from your terminal, install the
-[Solana CLI tool suite](https://docs.solana.com/cli/install-solana-cli-tools) on
-your local system:
+Just because rust compiles and builds your software into binary that can run for the computer architecture we specify, we need to install some OS level dependencies on our machine. 
 
-```bash
-sh -c "$(curl -sSfL https://release.solana.com/stable/install)"
+### Let’s start with Windows 8/10/11
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:62]: Windows 8 went out of support on January 12, 2016

File: content/guides/getstarted/setup-local-development.md [hunk 4]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
...
+
+There are two ways to get started on windows, 
+
+1.  build on windows
+2.  build on windows but with WSL
+
+WSL is windows subsystem for linux, which translates Linux system calls into Windows system calls using a compatibility layer, **for running Linux binary executables natively** and it runs Linux processes alongside the Windows kernel in a lightweight virtual machine.
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:69]: Might be too detailed. I just say "... allows you to run Linux software easily on Windows using a lightweight VM that instantly starts when you need it"

File: content/guides/getstarted/setup-local-development.md [hunk 5]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
...
+    
+    ![setup—-windows.exe](/assets/guides/setup-local-environment/setup---rustup-windows.png)
+    
+---
+
+### Build Dependencies for WSL
+
+1. Install WSL in your system, make sure to restart your computer when installing is done.
+    
+    [Install WSL](https://learn.microsoft.com/en-us/windows/wsl/install)
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:93]: Good stuff - better to link to current instructions like you're doing than to maintain our own, out of date, set of instructions.

File: content/guides/getstarted/setup-local-development.md [hunk 6]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
...
+    
+2. Follow this tutorial to open VSCode in WSL 
+    
+    [](https://code.visualstudio.com/docs/remote/wsl-tutorial)
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:97]: Missing link text

File: content/guides/getstarted/setup-local-development.md [hunk 7]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
...
+
+Installing dependencies in linux is same as windows with WSL.
+
+```jsx
+sudo apt-get install -y \
+    build-essential \
+    pkg-config \
+    libudev-dev llvm libclang-dev \
+    protobuf-compiler libssl-dev
+```
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:123]: That's not `jsx`, it's `bash`

File: content/guides/getstarted/setup-local-development.md [hunk 8]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
...
+xcode-select -p
+```
+
+If you don’t see a path returned, you need to install the cli tools.
+
+There are three ways to install xcode cli tools
+
+1. Install tools via terminal
+    
+    ```jsx
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:144]: Again this is `bash` (it's probably zsh on a mac but `bash` syntax highlighting works)

File: content/guides/getstarted/setup-local-development.md [hunk 9]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
...
+    xcode-select --install
+    ```
+    
+2. Download installer and install it with graphical interface
+    
+    [](https://developer.apple.com/download/all/)
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:150]: Missing link text

File: content/guides/getstarted/setup-local-development.md [hunk 10]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
...
-## Setup a localhost blockchain cluster
+---
+
+## 3. Install Solana CLI
+
+To interact with a Solana cluster (bunch of big state synced servers), we will use its command-line interface, also known as the CLI. We use the command-line because it is the first place the Solana core team deploys new functionality. The command-line interface is not necessarily the easiest to use, but it provides the most direct, flexible, and secure access to your Solana accounts (storage units on solana).
+
+### for windows
+
+1. Open terminal as administrator, copy and paste this command in your cmd, this downloads solana toolchain into a temporary location on C drive.
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:205]: Windows uses powershell as the default shell, also I think Windows may include curl now. Not sure `cmd` is necessary.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Heya! I saw this and thought I'd spend a few minutes checking it out. There's a few minor things to fix but this looks good once they're resolved. Get a review from a DevRel person too!

File: content/guides/getstarted/setup-local-development.md [hunk 11]
```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
...
+
+---
+
+Congrats, we have installed system dependencies and build essentials required to build software.
+
+---
+
+## 2. Install rust
+
+Rust is a multi-paradigm, general-purpose programming language that emphasizes performance, type safety, and concurrency. Or let’s just say its a coool language in boomer world.
```
[the remark this answers is not in the document]
Comment (heyAyushh) [reply content/guides/getstarted/setup-local-development.md:168]: I am sorry for this language, earlier I thought It was going on my blog, hence need a lot of reviews before.

**Comment by nickfrosty** [comment]
> I don't think Windows is supported, my bad, should I remove it from this guide?
> 
> I tried it on windows 10, while the cli was installed I couldn't get a test-validator running, Idk if one of us can confirm it.

yes, windows is not currently functional. Please remove that portion from this guide. 

Also, run the prettier formatting so all actions pass: `npm run prettier:fix`

**Comment by heyAyushh** [comment]
Closing this PR because I have contribute with the main branch, 
I will open new PR for this with a different branch

**Pull request closed without merging**